tests/conftest.py

```diff
@@ -28,48 +34,41 @@ def default_bbr() -> RuleCollection:
     return RuleCollection.default_bbr()
 
 
-class BaseRuleTest(unittest.TestCase):
+@pytest.fixture(scope="class")
+def rule_data(request):
+    global RULE_LOADER_FAIL, RULE_LOADER_FAIL_MSG
+    if not RULE_LOADER_FAIL:
+        try:
+            rc = default_rules()
+            rc_bbr = default_bbr()
+            request.cls.all_rules = rc.rules
+            request.cls.rule_lookup = rc.id_map
+            request.cls.production_rules = rc.filter(production_filter)
+            request.cls.bbr = rc_bbr.rules
+            request.cls.deprecated_rules: DeprecatedCollection = rc.deprecated
+        except Exception as e:
+            RULE_LOADER_FAIL = True
+            RULE_LOADER_FAIL_MSG = str(e)
+
+
+@pytest.mark.usefixtures("rule_data")
+class TestBaseRule:
```

Comment:
With all the wonderful features of pytests and fixtures, do we need a base class to be inherited everywhere else. I think we can avoid with altogether now?

Reply:
Discussed on DED call. Tabling this for another PR and task rather than including it in this PR, which is already rather large.